src/clr-angular/data/datagrid/providers/table-height.service.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {ElementRef, Injectable} from "@angular/core";
+
+/**
+ * @description
+ * Internal datagrid service that holds a reference to the clr-dg-table element.
+ */
+@Injectable()
+export class TableHeightService {
+    public tableRef: ElementRef;
+
+    set tableElement(table: ElementRef) {
+        this.tableRef = table;
+    }
```

Comment:
Could you add a `getHeight()` method or `get height()` setter to this class? It seems like the only way you're using this everywhere is by chaining to the native elements and computed height, so you should just expose directly a numeric height here.

On top of that, you have a setter on a public property, which is pretty useless in Typescript/Javascript. 😛 
You should just remove the setter.